webui/react/src/components/ResourceChart.tsx

```diff
@@ -83,25 +90,35 @@ const genPlotInfo = (title: string, resources: Resource[]): PlotInfo | null => {
   };
 };
 
-const SlotChart: React.FC<Props> = ({ title, resources, ...rest }: Props) => {
-  const [ oldPlotInfo, setOldPlotInfo ] = useState<PlotInfo | null>(null);
+const SlotChart: React.FC<Props> = (props: Props) => {
+  const chartRef = useRef<HTMLDivElement>(null);
+  const [ id ] = useState(generateAlphaNumeric());
+  const [ oldPlotInfo, setOldPlotInfo ] = useState<PlotInfo>(genPlotInfo(props.title, []));
 
   const plotInfo = useMemo(() => {
-    const newPlotInfo = genPlotInfo(title, resources || []);
+    const newPlotInfo = genPlotInfo(props.title, props.resources || []);
     if (JSON.stringify(newPlotInfo) === JSON.stringify(oldPlotInfo)) return oldPlotInfo;
     setOldPlotInfo(newPlotInfo);
     return newPlotInfo;
-  }, [ oldPlotInfo, resources, title ]);
+  }, [ oldPlotInfo, props.resources, props.title ]);
+
+  const renderPlot = useCallback(async (
+    elementId: string,
+    pInfo: PlotInfo,
+  ) => {
+    const args: PlotArguments = [ elementId, pInfo.data, pInfo.layout, pInfo.config ];
+    await Plotly.react.apply(null, args);
+  }, [ id, plotInfo, props.resources ]);
 
-  if (plotInfo === null) return <React.Fragment />;
+  useEffect(() => {
+    renderPlot(id, plotInfo);
+  }, [ id, plotInfo, props.resources ]);
 
   return (
-    <Plot
-      {...rest}
-      config={{ displaylogo: false, displayModeBar: false, responsive: false }}
-      data={plotInfo.data}
-      layout={plotInfo.layout}
-    />
+    // <div className={css.base}>
+    <div>
+      <div id={id} ref={chartRef} />
+    </div>
```

Comment:
**chore:** minor list of cleanup
- drop the comment
- remove the empty `div` wrapping the chart `div`
- remove `ref={chartRef}`, looks like the reference is not used anywhere